spec/models/searchgov_url_spec.rb

```diff
@@ -150,6 +162,69 @@
         fetch
       end
 
+      context 'when searchgov_document save functionality is flagged off' do
+        before do
+          ENV['SEARCHGOV_DOCUMENT_NO_SAVE'] = 'true'
+        end
+
+        after do
+          ENV['SEARCHGOV_DOCUMENT_NO_SAVE'] = nil
+        end
+
+        it 'does not save a document to searchgov_documents' do
+          expect { fetch }.not_to change { SearchgovDocument.count }
+        end
+      end
+
+      context 'when no searchgov_document exists' do
+        it 'saves the document to searchgov_documents' do
+          expect { fetch }.to change { SearchgovDocument.count }.
+            from(0).to(1)
+        end
+
+        it 'saves html content in searchgov_documents' do
+          fetch
+          doc = SearchgovDocument.find_by(searchgov_url_id: searchgov_url.id)
+          expect(doc.body).to include('This is my content.')
```

Comment:
I would simplify this a bit, but also include the tags to make it clearer that the HTML is actually what's being stored (as opposed to the parsed content):
```ruby
doc = searchgov_url.searchgov_document
expect(doc.body).to include('<p>This is my content.</p>')
```
In fact, it would DRY things up to assign the `searchgov_document` for use anywhere in the fetching specs:
```ruby
let(:searchgov_document) { searchgov_url.searchgov_document }
```
That will allow you to replace all the instances of:
```ruby
SearchgovDocument.find_by(searchgov_url_id: searchgov_url.id)
```
with:
```ruby
searchgov_document
```

Reply:
👍 Major improvement.